Added support for datepicker field

vanmil opened the pull request:
This change adds support for the Datepicker field type in the syntax parser. The available datepicker options (mode, format, minDate, maxDate, firstDay) discussed [here](https://octobercms.com/docs/backend/forms#widget-datepicker) are working correctly.

LukeTowers commented on src/Parse/Syntax/Parser.php:
```diff
@@ -218,6 +218,13 @@ protected function evalTwigViewField($field, $params, $prefix = null)
                 $result = '{% if ' . $field . ' %}' . $params['_content'] . '{% endif %}';
                 break;
+            case 'datepicker':
+                if ($params['mode'] === 'date'){
```
This doesn't actually properly support the `mode` option. The `mode` option supports three values, `date`, `time`, and `datetime`. Up to you if that makes more sense with a bunch of elseifs or a select case setup.

LukeTowers commented:
This pull request has been made to the wrong branch. Please review [the contributing guidelines](https://github.com/octobercms/october/blob/master/CONTRIBUTING.md#pull-requests) as all PRs need to be made to the develop branch.

I'll fix it for you this time, but please ensure you make any future PRs to the develop branch, not the master branch.

vanmil commented:
I changed it from an if construction to switch case construction and also added the datetime case.

For the `ignoreTimezone` option, it doesn't not work yet, because I found a more generic bug. The parser does not parses boolean strings to real booleans. I made a separate pull request for this, because it is a more generic thing which also has implications for other field types besides the datepicker, see #283  for that request. When that request is added, the `ignoreTimezone` option does work.
